gson/src/test/java/com/google/gson/FieldNamingPolicyTest.java

```diff
@@ -0,0 +1,78 @@
+package com.google.gson;
+
+import org.junit.Test;
+import org.junit.runners.Parameterized.Parameters;
+
+import static org.junit.Assert.assertEquals;
+
+public class FieldNamingPolicyTest {
+	
+
+	@Test
+	public void testUpperCaseFirstLetterNonUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("someFieldName", "SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("SomeFieldName", "SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderscoreAndNonUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_someFieldName", "_SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderScoreAndWithUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_SomeFieldName", "_SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderscoreAndNumberAndNonUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_999someFieldName", "_999SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderScoreAndNumberWithUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_333SomeFieldName", "_333SomeFieldName");
+	}
+
+	@Test
+	public void testUpperCaseFirstLetter1LowercaseLetter() {
+		verifyUpperCaseFirstLetter("x", "X");
+	}
+
+	@Test
+	public void testUpperCaseFirstLetter1UppercaseLetter() {
+		verifyUpperCaseFirstLetter("X", "X");
+	}
+
+	@Test
+	public void testUpperCaseFirstAllNumbers() {
+		verifyUpperCaseFirstLetter("555", "555");
+	}
+
+	@Test
+	public void testUpperCaseFirstJustUnderscore() {
+		verifyUpperCaseFirstLetter("_", "_");
+	}
+
+	@Test
+	public void testUpperCaseBlank() {
+		verifyUpperCaseFirstLetter(" ", " ");
```

Comment:
you can probably collapse all of these tests into a single test method. It’s extremely unlikely we’ll ever want to run these tests independently, or that they’ll fail independently. That way if/when we ever want to add tests for more policies, we don’t need to write thousands of lines of code.